markdown/guide/lazy-loading.md

```diff
@@ -57,24 +57,161 @@ Note that route serializers are unique to Engines and won't work in normal appli
 
 #### Loading / Error Substates
 
-The loading and error substates work in a similar fashion to [substates in a normal Ember app](https://guides.emberjs.com/v2.9.0/routing/loading-and-error-substates/). The only difference is that lazily loaded Engines will enter a loading state while the assets for the Engine are loaded and can enter an error state when an asset fails to load.
+The loading and error substates work in a similar fashion to [substates in a normal Ember app](https://guides.emberjs.com/release/routing/loading-and-error-substates/). The only difference is that lazily loaded Engines will enter a loading state while the assets for the Engine are loaded and can enter an error state when an asset fails to load.
 
-### Build Behavior
+### Built Engine Output
+
+### Eager Engines
+
+Eager engines are built approximately the same as existing addons. Differences
+are limited to consolidating the namespace of `app` code inside of an engine.
+Into the engine's namespace instead of the host application.
+
+Beyond that it adds in a configuration module for the engine, and nothing else. 
+It is a remarkably straightforward process.
+
+ ### Lazy Engines
 
 Lazy engines are built in the same way as eager engines, but their assets are not combined back into the host application's `vendor.js` file. This means that they are run through a separate and unique build process from what a default addon will go through, though it reaches out to the upstream implementation in Ember CLI where possible.
 
-A lazy engine's output (lazy-engine) looks like this:
+A lazy engine's output (`lazy-engine`) looks like this:
 
 ```
 dist
+├── assets
+│   ├── host-application.css
+│   ├── host-application.js
+│   ├── vendor.css
+│   └── vendor.js
 ├── engines-dist
 │   └── lazy-engine
+|       ├── config
+│       |   ├── enviroment.js
 │       ├── assets
 │       │   ├── engine-vendor.css
 │       │   ├── engine-vendor.js
 │       │   ├── engine.css
 │       │   └── engine.js
 │       └── public-asset.jpg
+├── asset-manifest.json
+├── crossdomain.xml
+├── index.html
+└── robots.txt
+```
+
+#### `/routes.js`
+
+ The `routes.js` file and anything it `import`s must be present at boot time of
+the host application. It will be bundled into the host application's `vendor.js`
+file. This location should be considered `undefined` behavior and should not be
+relied upon as it may change in the future.
+
+ Its module name inside of the host application will be `lazy-engine/routes`. Any
+`import`s will also be in the `lazy-engine` module path.
+
+ #### `/app`
+
+ Assets in this folder don't make sense and will be ignored as they break the
+isolation guarantees of engines.
```

Comment:
I'm not positive that app assets are actually ignored. I agree it's a bad idea to use them.

Reply:
@dgeb sure, `app` assets don't are ignored, but our documentation teaches to reexport components in the `app/`.

What do we do in this case?

Reply:
fixed